tests/deploy/e2e/test_autostarted.py

```diff
@@ -315,3 +360,779 @@ async def test_autostart_clear_agent_venv_on_delete(
+b = minimalwaitingmodule::Sleep(name="test_sleep2", agent="agent1", time_to_sleep=5)
+c = minimalwaitingmodule::Sleep(name="test_sleep3", agent="agent1", time_to_sleep=5)
+""",
+        autostd=True,
+    )
+
+    # Now, let's deploy some resources
+    version, res, status = await snippetcompiler.do_export_and_deploy(include_status=True)
+    result = await client.release_version(environment, version, push=False)
+    assert result.code == 200
+
+    async def check_resource_in_state(expected_resources: int = 1, state: str = "deployed") -> bool:
+        result = await client.resource_list(environment, deploy_summary=True)
+        assert result.code == 200
+        summary = result.result["metadata"]["deploy_summary"]
+        nr_res_desired_state = summary["by_state"][state]
+        return nr_res_desired_state == expected_resources
+
+    # Wait for at least one resource to be deployed
+    await retry_limited(check_resource_in_state, timeout=10)
+
+    # Make sure the children of the scheduler are consistent
+    await wait_for_consistent_children(
+        current_pid=current_pid,
+        should_scheduler_be_defined=True,
+        should_fork_server_be_defined=True,
+        executor_to_be_defined=1,
+    )
+
+    # Retrieve the current processes, we should have more processes than `start_children`
+    state_after_deployment = construct_scheduler_children(current_pid)
+    for children in state_after_deployment.children:
+        assert children.is_running()
+
+    # The number of resources for this version should match
+    result = await client.list_versions(tid=environment)
+    assert result.code == 200
+    assert len(result.result["versions"]) == 1
+    assert result.result["versions"][0]["total"] == 3
+
+    # Let's check the agent table and check that agent1 is present and not paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: False, "agent1": False})
+
+    # Now let's pause agent1
+    result = await client.agent_action(tid=environment, name="agent1", action=AgentAction.pause.value)
+    assert result.code == 200
+
+    # Let's recheck the agent table and check that agent1 is present and paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: False, "agent1": True})
+
+    # Let's also check if the state of resources are consistent with what we expect: one resource still needs to be
+    # deployed
+    await retry_limited(check_resource_in_state, timeout=6, expected_resources=2)
+    result = await client.resource_list(environment, deploy_summary=True)
+    assert result.code == 200
+    summary = result.result["metadata"]["deploy_summary"]
+    assert summary["total"] == 3, f"Unexpected summary: {summary}"
+    assert summary["by_state"]["available"] == 1, f"Unexpected summary: {summary}"
+    assert summary["by_state"]["deployed"] == 2, f"Unexpected summary: {summary}"
+
+    # Let's wait for the executor to die
```

Comment:
Why does this kill the executor ?

Reply:
It's not being killed but cleaned up. I refactored these comments